src/context.rs

```diff
@@ -668,6 +756,92 @@ mod tests {
         }
     }
 
+    mod before_all {
```

Comment:
thanks for your new tests, I have missed this flow in my code.

In order to check the correct properties, I suggest these tests:
```rust

    mod before_all {
        pub use super::*;

        #[test]
        fn can_be_called_inside_describe() {
            rdescribe("a root", |ctx| { ctx.before_all(|| {}) });
        }

        #[test]
        fn can_be_called_inside_nested_describe() {
            rdescribe("a root", |ctx| {
                ctx.describe("a nested describe", |ctx| {
                    ctx.before_all(|| {})
                });
            });
        }

        #[test]
        fn it_will_be_called_before_any_it_of_its_group() {
            use std::sync::atomic::{AtomicUsize, Ordering};
            let ran_counter = &mut AtomicUsize::new(0);

            rdescribe("root", |ctx| {
                ctx.it("should see the before_all hook", ||
                       1 == ran_counter.load(Ordering::SeqCst));
                ctx.before_all(|| { ran_counter.fetch_add(1, Ordering::SeqCst); });
            })
        }

        #[test]
        fn it_will_be_called_only_once_in_its_group() {
            use std::sync::atomic::{AtomicUsize, Ordering};
            let ran_counter = &mut AtomicUsize::new(0);

            rdescribe("root", |ctx| {
                ctx.before_all(|| { ran_counter.fetch_add(1, Ordering::SeqCst); });
                ctx.it("should see the before_all hook", ||
                       1 == ran_counter.load(Ordering::SeqCst));
                ctx.it("should see the before_all hook no more", ||
                       1 == ran_counter.load(Ordering::SeqCst));
            })
        }
    }
```